File: test_connections.py
#!/usr/bin/env python3
"""
Test script to verify Azure and MySQL connections
"""

import os
import sys
import asyncio
from pathlib import Path
from dotenv import load_dotenv

# Load environment variables
load_dotenv()

async def test_azure_services():
    """Test Azure service connections"""
    print("🔍 Testing Azure Services...")
    
    try:
        # Test Azure Storage
        from azure.storage.blob import BlobServiceClient
        storage_conn = os.getenv('AZURE_STORAGE_CONNECTION_STRING')
        if storage_conn:
            blob_client = BlobServiceClient.from_connection_string(storage_conn)
            # Try to list containers (this will test the connection)
            containers = list(blob_client.list_containers())
            print("✅ Azure Blob Storage: Connected successfully")
        else:
            print("❌ Azure Blob Storage: No connection string found")
    except Exception as e:
        print(f"❌ Azure Blob Storage: {e}")
    
    try:
        # Test Azure Form Recognizer
        from azure.ai.formrecognizer import DocumentAnalysisClient
        from azure.core.credentials import AzureKeyCredential
        
        endpoint = os.getenv('AZURE_COGNITIVE_SERVICES_ENDPOINT')
        key = os.getenv('AZURE_COGNITIVE_SERVICES_KEY')
        
        if endpoint and key:
            client = DocumentAnalysisClient(endpoint=endpoint, credential=AzureKeyCredential(key))
            print("✅ Azure Form Recognizer: Client initialized successfully")
        else:
            print("❌ Azure Form Recognizer: Missing endpoint or key")
    except Exception as e:
        print(f"❌ Azure Form Recognizer: {e}")
    
    try:
        # Test Azure Text Analytics
        from azure.ai.textanalytics import TextAnalyticsClient
        from azure.core.credentials import AzureKeyCredential
        
        endpoint = os.getenv('AZURE_TEXT_ANALYTICS_ENDPOINT')
        key = os.getenv('AZURE_TEXT_ANALYTICS_KEY')
        
        if endpoint and key:
            client = TextAnalyticsClient(endpoint=endpoint, credential=AzureKeyCredential(key))
            # Test with a simple text
            test_text = ["This is a test document for real estate processing."]
            result = client.recognize_entities(documents=test_text)
            print("✅ Azure Text Analytics: Connected and working")
        else:
            print("❌ Azure Text Analytics: Missing endpoint or key")
    except Exception as e:
        print(f"❌ Azure Text Analytics: {e}")

def test_mysql_connection():
    """Test MySQL database connection"""
    print("\n🔍 Testing MySQL Database...")
    
    try:
        import pymysql
        
        connection = pymysql.connect(
            host=os.getenv('MYSQL_HOST'),
            port=int(os.getenv('MYSQL_PORT')),
            user=os.getenv('MYSQL_USER'),
            password=os.getenv('MYSQL_PASSWORD'),
            database=os.getenv('MYSQL_DATABASE'),
            charset='utf8mb4'
        )
        
        cursor = connection.cursor()
        cursor.execute("SELECT VERSION()")
        version = cursor.fetchone()
        print(f"✅ MySQL Database: Connected successfully (Version: {version[0]})")
        
        # Test if our table exists
        cursor.execute("SHOW TABLES LIKE 'documents'")
        table_exists = cursor.fetchone()
        if table_exists:
            print("✅ Documents table: Exists")
        else:
            print("⚠️ Documents table: Not found (will be created on first run)")
        
        cursor.close()
        connection.close()
        
    except Exception as e:
        print(f"❌ MySQL Database: {e}")

async def test_document_processing():
    """Test the document processing pipeline"""
    print("\n🔍 Testing Document Processing Pipeline...")
    
    try:
        # Import our services
        from app.services.ocr_service import OCRService
        from app.services.nlp_service import NLPService
        from app.services.storage_service import StorageService
        from app.services.document_processor import DocumentProcessor
        
        # Initialize services
        ocr_service = OCRService()
        nlp_service = NLPService()
        storage_service = StorageService()
        document_processor = DocumentProcessor(ocr_service, nlp_service, storage_service)
        
        print("✅ All services initialized successfully")
        
        # Create a test document
        test_content = """
        REAL ESTATE LEASE AGREEMENT
        
        Property Address: 123 Main Street, Test City, TC 12345
        
        Landlord: John Smith
        Phone: (555) 123-4567
        
        Tenant: Jane Doe
        Phone: (555) 987-6543
        
        Monthly Rent: $1,500.00
        Security Deposit: $1,500.00
        Lease Start Date: January 1, 2024
        Lease End Date: December 31, 2024
        
        Property Details:
        Type: Apartment
        Bedrooms: 2
        Bathrooms: 1
        Square Footage: 900 sq ft
        """
        
        # Create test file
        test_file = Path("test_document.txt")
        test_file.write_text(test_content)
        
        # Process the document
        result = await document_processor.process_document(str(test_file))
        
        if result.get('status') == 'success':
            print("✅ Document processing: Successful")
            print(f"   - Extracted {len(result.get('raw_text', ''))} characters")
            print(f"   - Confidence score: {result.get('confidence_score', 0)}")
            
            extracted_data = result.get('extracted_data', {})
            if extracted_data:
                print("   - Extracted fields:")
                for key, value in extracted_data.items():
                    if value and key != 'property_details':
                        print(f"     • {key}: {value}")
        else:
            print(f"❌ Document processing: Failed - {result.get('error', 'Unknown error')}")
        
        # Clean up
        if test_file.exists():
            test_file.unlink()
            
    except Exception as e:
        print(f"❌ Document processing: {e}")

async def main():
    """Main test function"""
    print("=" * 60)
    print("Real Estate Document Processor - Connection Tests")
    print("=" * 60)
    
    # Test Azure services
    await test_azure_services()
    
    # Test MySQL
    test_mysql_connection()
    
    # Test document processing
    await test_document_processing()
    
    print("\n" + "=" * 60)
    print("🎉 Connection testing completed!")
    print("=" * 60)

if __name__ == "__main__":
    asyncio.run(main())

File: test-connections.js
const mysql = require('mysql2/promise');
const axios = require('axios');
const fs = require('fs').promises;
require('dotenv').config();

async function testConnections() {
  console.log('\n' + '='.repeat(60));
  console.log('🧪 Testing Azure and MySQL Connections');
  console.log('='.repeat(60));
  
  let allTestsPassed = true;
  
  // Test MySQL Database
  console.log('\n📊 Testing MySQL Database Connection...');
  try {
    console.log(`   Connecting to: ${process.env.MYSQL_HOST}:${process.env.MYSQL_PORT}`);
    console.log(`   Database: ${process.env.MYSQL_DATABASE}`);
    console.log(`   User: ${process.env.MYSQL_USER}`);
    
    const connection = await Promise.race([
      mysql.createConnection({
        host: process.env.MYSQL_HOST,
        port: process.env.MYSQL_PORT,
        user: process.env.MYSQL_USER,
        password: process.env.MYSQL_PASSWORD,
        database: process.env.MYSQL_DATABASE,
        connectTimeout: 30000,
        acquireTimeout: 30000,
        timeout: 30000
      }),
      new Promise((_, reject) => 
        setTimeout(() => reject(new Error('Connection timeout after 30 seconds')), 30000)
      )
    ]);
    
    const connection_old = await mysql.createConnection({
      host: process.env.MYSQL_HOST,
      port: process.env.MYSQL_PORT,
      user: process.env.MYSQL_USER,
      password: process.env.MYSQL_PASSWORD,
      database: process.env.MYSQL_DATABASE,
      connectTimeout: 30000,
      acquireTimeout: 30000,
      timeout: 30000
    });
    
    await connection.ping();
    console.log('✅ MySQL connection successful');
    
    // Test table creation
    await connection.execute(`
      CREATE TABLE IF NOT EXISTS test_table (
        id INT AUTO_INCREMENT PRIMARY KEY,
        test_data VARCHAR(255)
      )
    `);
    console.log('✅ MySQL table operations working');
    
    await connection.execute('DROP TABLE IF EXISTS test_table');
    await connection.end();
    
  } catch (error) {
    console.log('❌ MySQL connection failed:', error.message);
    allTestsPassed = false;
  }
  
  // Test Azure Form Recognizer
  console.log('\n🔍 Testing Azure Form Recognizer...');
  try {
    const response = await axios.get(
      `${process.env.AZURE_COGNITIVE_SERVICES_ENDPOINT}/formrecognizer/info?api-version=2023-07-31`,
      {
        headers: {
          'Ocp-Apim-Subscription-Key': process.env.AZURE_COGNITIVE_SERVICES_KEY
        }
      }
    );
    console.log('✅ Azure Form Recognizer connection successful');
  } catch (error) {
    console.log('❌ Azure Form Recognizer failed:', error.response?.data || error.message);
    allTestsPassed = false;
  }
  
  // Test Azure Text Analytics
  console.log('\n🧠 Testing Azure Text Analytics...');
  try {
    const response = await axios.post(
      `${process.env.AZURE_TEXT_ANALYTICS_ENDPOINT}/text/analytics/v3.1/entities/recognition/general`,
      {
        documents: [
          {
            id: "1",
            language: "en",
            text: "This is a test document for entity recognition."
          }
        ]
      },
      {
        headers: {
          'Ocp-Apim-Subscription-Key': process.env.AZURE_TEXT_ANALYTICS_KEY,
          'Content-Type': 'application/json'
        }
      }
    );
    console.log('✅ Azure Text Analytics connection successful');
  } catch (error) {
    console.log('❌ Azure Text Analytics failed:', error.response?.data || error.message);
    allTestsPassed = false;
  }
  
  // Test Azure Blob Storage
  console.log('\n☁️ Testing Azure Blob Storage...');
  try {
    // Simple connection test by parsing connection string
    const connectionString = process.env.AZURE_STORAGE_CONNECTION_STRING;
    if (connectionString && connectionString.includes('AccountName') && connectionString.includes('AccountKey')) {
      console.log('✅ Azure Blob Storage connection string is valid');
    } else {
      throw new Error('Invalid connection string format');
    }
  } catch (error) {
    console.log('❌ Azure Blob Storage configuration failed:', error.message);
    allTestsPassed = false;
  }
  
  // Summary
  console.log('\n' + '='.repeat(60));
  if (allTestsPassed) {
    console.log('🎉 All connection tests passed! Ready to start the server.');
    console.log('Run: npm start');
  } else {
    console.log('⚠️ Some tests failed. Please check your configuration.');
  }
  console.log('='.repeat(60));
}

testConnections().catch(console.error);